src/containers/ServiceAccountsDropdown/ServiceAccountsDropdown.js

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { Dropdown, DropdownSkeleton } from 'carbon-components-react';
+
+import {
+  getServiceAccounts,
+  isFetchingServiceAccounts,
+  getSelectedNamespace
+} from '../../reducers';
+import { fetchServiceAccounts } from '../../actions/serviceAccounts';
+
+class ServiceAccountsDropdown extends React.Component {
+  componentDidMount() {
+    this.props.fetchServiceAccounts();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { namespace } = this.props;
+    if (namespace !== prevProps.namespace) {
+      this.props.fetchServiceAccounts();
+    }
+  }
+
+  render() {
+    const { loading, ...dropdownProps } = this.props;
+    if (loading) {
+      return <DropdownSkeleton {...dropdownProps} />;
+    }
+    return <Dropdown {...dropdownProps} />;
```

Comment:
The Dropdown component truncates long options but does not display a tooltip. To allow users to correctly identify the options we made some additions to the NamespacesDropdown to add a `title` attribute on the options. That could should be refactored for reuse here.

https://github.com/tektoncd/dashboard/blob/fc9259d40e958dfe2d3f6bd8307c58a1aafaed10/src/containers/NamespacesDropdown/NamespacesDropdown.js#L22

see the `itemToElement` and `itemToString` props passed to the Dropdown component

Reply:
Ok, will do